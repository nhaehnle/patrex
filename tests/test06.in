Foo::Foo
	(A & a, B & b, bool c)
	:
	NS::Base(a, b),
	m_toggle_chat
		(INIT_BTN("menu_chat", "chat", toggle_chat, _("Chat"))),
	m_exit
		(INIT_BTN("menu_exit_game", "exit_replay", exit_btn, _("Exit Replay"))),
	m_save
		(INIT_BTN("menu_save_game", "save_game", save_btn, _("Save Game"))),
	m_toggle_options_menu
		(INIT_BTN("menu_options_menu", "options_menu",
					 toggle_options_menu, _("Options"))),
	m_toggle_statistics
		(INIT_BTN("menu_general_stats", "general_stats",
					 toggle_statistics, _("Statistics"))),
	m_toggle_minimap
		(INIT_BTN("menu_toggle_minimap", "minimap",
					 toggle_minimap, _("Minimap")))
{
	initializations();
}
