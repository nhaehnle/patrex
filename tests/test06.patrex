
define init ${id}|field| ( INIT_BTN( $.*, $.* $(, ${id}|slot| )|arg| , $.* ) )
define arg $|( $!(,)* ){init}

match : ${arg}+(,)[inits] { $<|block| $.* }
	forall inits
		if field
			insert block "\n\t{field}.sigclicked.connect(boost::bind(&Foo::{slot}, this));"
			erase arg
	insert block "\n"
