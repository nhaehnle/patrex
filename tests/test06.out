Foo::Foo
	(A & a, B & b, bool c)
	:
	NS::Base(a, b),
	m_toggle_chat
		(INIT_BTN("menu_chat", "chat", _("Chat"))),
	m_exit
		(INIT_BTN("menu_exit_game", "exit_replay", _("Exit Replay"))),
	m_save
		(INIT_BTN("menu_save_game", "save_game", _("Save Game"))),
	m_toggle_options_menu
		(INIT_BTN("menu_options_menu", "options_menu", _("Options"))),
	m_toggle_statistics
		(INIT_BTN("menu_general_stats", "general_stats", _("Statistics"))),
	m_toggle_minimap
		(INIT_BTN("menu_toggle_minimap", "minimap", _("Minimap")))
{
	m_toggle_chat.sigclicked.connect(boost::bind(&Foo::toggle_chat, this));
	m_exit.sigclicked.connect(boost::bind(&Foo::exit_btn, this));
	m_save.sigclicked.connect(boost::bind(&Foo::save_btn, this));
	m_toggle_options_menu.sigclicked.connect(boost::bind(&Foo::toggle_options_menu, this));
	m_toggle_statistics.sigclicked.connect(boost::bind(&Foo::toggle_statistics, this));
	m_toggle_minimap.sigclicked.connect(boost::bind(&Foo::toggle_minimap, this));

	initializations();
}
